src/twisted/internet/_resolver.py

```diff
@@ -26,14 +39,25 @@
 from twisted.internet.defer import Deferred
 from twisted.internet.error import DNSLookupError
 from twisted.internet.interfaces import (
+    IAddress,
     IHostnameResolver,
     IHostResolution,
+    IReactorThreads,
     IResolutionReceiver,
     IResolverSimple,
 )
 from twisted.internet.threads import deferToThreadPool
 from twisted.logger import Logger
 from twisted.python.compat import nativeString
+from twisted.python.failure import Failure
+
+if TYPE_CHECKING:
+    from twisted.python.runtime import platform
+
+    if platform.supportsThreads():
+        from twisted.python.threadpool import ThreadPool
+    else:
+        ThreadPool = object  # type: ignore[misc, assignment]
```

Comment:
While this might *work*, it's kind of incoherent and very confusing to read. I think what you want is simply this:

```suggestion
    from twisted.python.threadpool import ThreadPool
```

The code in that other file is a mistake, so let's not repeat it here.

Reply:
I double checked this change is fine with mypy, so let's commit it! 👍